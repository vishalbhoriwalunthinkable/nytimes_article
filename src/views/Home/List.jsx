import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { List } from "@material-ui/core";
import PropTypes from "prop-types";
import ListItem from "../../components/ListItem";
import ContentWrap from "../../components/ContentWrap/ContentWrap";

const useStyles = makeStyles((theme) => ({
  root: {
    width: "100%",
    maxWidth: "850px",
    margin: "0 auto",
    alignItems: "center",
    wordBreak: "break-text",
    justifyContent: "center",
    backgroundColor: theme.palette.background.paper,
  },
}));

export default function FolderList({ info }) {
  const classes = useStyles();
  const { isFetching, isError, data = [] } = info;

  return (
    <ContentWrap isFetching={isFetching} isError={isError}>
      <List className={classes.root}>
        {data && data.length ? (
          data.map((val, index) => <ListItem key={index * 2 + 1} data={val} />)
        ) : (
          <div>Item not found !</div>
        )}
      </List>
    </ContentWrap>
  );
}

FolderList.propTypes = {
  info: PropTypes.object.isRequired,
};
