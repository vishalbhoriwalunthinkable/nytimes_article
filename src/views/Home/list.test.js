import React from "react";
import renderer from "react-test-renderer";

import List from "./List";

it("list renders when there are no items", () => {
  const info = {
    isFetching: false,
    isError: false,
    data: [],
  };
  const tree = renderer.create(<List info={info} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("list renders when there are one item", () => {
  const info = {
    isFetching: false,
    isError: false,
    data: [{ title: "Hello", content: "content" }],
  };
  const tree = renderer.create(<List info={info} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("list renders when there is error", () => {
  const info = {
    isFetching: false,
    isError: true,
    data: [],
  };
  const tree = renderer.create(<List info={info} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("list renders when is fetcing true", () => {
  const info = {
    isFetching: true,
    isError: false,
    data: [],
  };
  const tree = renderer.create(<List info={info} />).toJSON();
  expect(tree).toMatchSnapshot();
});
