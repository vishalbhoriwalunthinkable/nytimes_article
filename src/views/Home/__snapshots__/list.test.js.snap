// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list renders when is fetcing true 1`] = `
<div
  className="makeStyles-root-2"
  data-testid="loading-component"
  style={
    Object {
      "margin": "0px auto",
      "minHeight": "200px",
      "width": "10%",
    }
  }
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`list renders when there are no items 1`] = `
<ul
  className="MuiList-root makeStyles-root-1 MuiList-padding"
>
  <div>
    Item not found !
  </div>
</ul>
`;

exports[`list renders when there are one item 1`] = `
<ul
  className="MuiList-root makeStyles-root-1 MuiList-padding"
>
  <a
    className="makeStyles-link-7"
    rel="noreferrer"
    target="_blank"
  >
    <li
      className="MuiListItem-container"
    >
      <div
        className="MuiListItem-root makeStyles-listItem-5 MuiListItem-gutters MuiListItem-alignItemsFlexStart MuiListItem-secondaryAction"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiListItemText-root makeStyles-listitemtext-8 MuiListItemText-multiline"
          style={
            Object {
              "textAlign": "justify",
              "width": "210px",
            }
          }
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            <div
              className="ellipsis"
            >
              NA
            </div>
          </span>
          <p
            className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <div
              style={
                Object {
                  "display": "inline",
                  "justifyContent": "space-between",
                }
              }
            >
              <div />
              <div
                className="makeStyles-date-6"
                color="textPrimary"
                variant="body2"
              >
                <div
                  style={
                    Object {
                      "display": "flex",
                      "width": "fit-content",
                    }
                  }
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zM7 10h5v5H7z"
                    />
                  </svg>
                  <div>
                    NA
                  </div>
                </div>
              </div>
            </div>
          </p>
        </div>
      </div>
      <div
        className="MuiListItemSecondaryAction-root"
        style={
          Object {
            "right": "0",
          }
        }
      >
        <button
          aria-label="comments"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </span>
        </button>
      </div>
    </li>
  </a>
</ul>
`;

exports[`list renders when there is error 1`] = `
<div
  className="ContentWrap"
>
  Something went wrong...
</div>
`;
